### PR TITLE
Add simple TabPFN example

### PR DESCRIPTION
**Pull request by zechengz**
- Fixed one doc issue
- Add a simple script to some classification tasks
  - Support minibatch of training data for the prediction with minibatch of test data
  - Titanic - Accuracy: 0.8202
  - Mushroom - Accuracy: 1.0000(ok)
  - ForestCoverType - Accuracy: 0.8205 (ok, outperforms some model)
  - KDDCensusIncome - Accuracy: 0.9537 (ok)

### CODE REVIEW

File: examples/tabpfn_classification.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    for test_data in tqdm(test_loader):
+        X_train = []
+        X_test = []
+        train_data = next(iter(train_loader))
+        for stype in train_data.stypes:
+            X_train.append(train_data.feat_dict[stype])
+            X_test.append(test_data.feat_dict[stype])
+        X_train = torch.cat(X_train, dim=1)
+        X_test = torch.cat(X_test, dim=1)
+        clf = TabPFNClassifier()
```
Comment (zechengz) [inline examples/tabpfn_classification.py:64]: Seems that we need to reinitialize the `TabPFNClassifier` each time...

**Review by yiweny — APPROVED** [review]

File: examples/tabpfn_classification.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+from torch_frame.data import DataLoader
+from torch_frame.datasets import (
+    ForestCoverType,
+    KDDCensusIncome,
+    Mushroom,
+    Titanic,
+)
+
+parser = argparse.ArgumentParser()
```
Comment (akihironitta) [inline examples/tabpfn_classification.py:18]: nit:
```suggestion
parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

**Review by akihironitta — APPROVED** [review]

File: examples/tabpfn_classification.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+train_dataset, test_dataset = dataset[:0.9], dataset[0.9:]
+train_tensor_frame = train_dataset.tensor_frame
+test_tensor_frame = test_dataset.tensor_frame
+train_loader = DataLoader(
+    train_tensor_frame,
+    batch_size=args.train_batch_size,
+    shuffle=True,
+)
+X_train = []
+train_data = next(iter(train_loader))
```
Comment (weihua916) [inline examples/tabpfn_classification.py:55]: why only sample random minibatch this way?

Comment (weihua916) [reply examples/tabpfn_classification.py:55]: Got it, you probably don't need data loader. Just use index select over the original tensor frame.
